airbyte-webapp/src/pages/SettingsPage/pages/NotificationPage/components/WebHookForm.tsx

```diff
@@ -99,76 +88,179 @@ const WebHookForm: React.FC<WebHookFormProps> = ({ webhook, onSubmit, successMes
+                  </Field>
+                  {errorMessage ? (
+                    <Text className={styles.webhookErrorMessage} size="sm">
+                      <FormattedMessage id="form.someError" />
+                    </Text>
+                  ) : null}
+                </Cell>
+                <Cell className={styles.testButtonCell}>
+                  <Tooltip
+                    className={styles.tooltip}
+                    placement="top"
+                    control={
+                      <Button
+                        className={styles.testButton}
+                        size="sm"
+                        type="button"
+                        variant="secondary"
+                        isLoading={formAction.test}
+                        disabled={!values.webhook || !!errors.webhook || formAction.save}
+                        onClick={() => webhookChange(WebhookAction.Test, values)}
+                      >
+                        <FormattedMessage id="settings.test" />
+                      </Button>
+                    }
+                  >
+                    <FormattedMessage id="settings.webhookTestText" />
+                  </Tooltip>
+                </Cell>
+              </Row>
+              <Row className={styles.notificationSettingsLabelRow}>
+                <Cell className={styles.notificationSettingsLabelCell}>
+                  <Label>
+                    <FormattedMessage id="settings.syncNotifications.label" />
+                  </Label>
+                </Cell>
+              </Row>
+              <Row className={styles.notificationSettingsRow}>
+                <Cell className={styles.notificationSettingsCell}>
+                  <Field name="sendOnFailure">
+                    {({ field }: FieldProps<boolean>) => (
+                      <LabeledSwitch
+                        className={styles.sendOnFailure}
+                        name={field.name}
+                        checked={field.value}
+                        onChange={field.onChange}
+                        label={<FormattedMessage id="settings.sendOnFailure" />}
+                      />
+                    )}
+                  </Field>
+                  <Field name="sendOnSuccess">
+                    {({ field }: FieldProps<boolean>) => (
+                      <LabeledSwitch
+                        name={field.name}
+                        checked={field.value}
+                        onChange={field.onChange}
+                        label={<FormattedMessage id="settings.sendOnSuccess" />}
+                      />
+                    )}
+                  </Field>
+                </Cell>
+              </Row>
+            </Content>
+          </SettingsCard>
+          <div className={styles.action}>
+            <Button
+              type="submit"
+              size="sm"
+              isLoading={formAction.save}
+              disabled={!dirty || !!errors.webhook || formAction.test}
```

Comment:
We need to clarify the behavior of such a form. 

Currently, if the user entered a **valid** URL, he can test it (button `Test`) and save changes (button `Save changes`). Because the form is valid. Users can't test or save changes if an entered string does not match the URL format. Also, the user can test his webhook without saving.